### ISSUE TITLE
Incorrect permissions on *.egg-info directory

### ISSUE THREAD

**Issue by dan-santana** [pypa/setuptools#1328]
It seems that the `setup.py install` command doesn't set the permissions of the `<package>.egg-info/*` files when they already exist, relying on what is already set. As noted in https://github.com/sshuttle/sshuttle/issues/217#issuecomment-381378684, it seems many other projects suffered from the same "bug" in the last years, relying on the packager to set the permissions manually to 644 before uploading to PyPI.

Relying on the umask of the user during the package tarball creation is too error-prone, so maybe permissions should always be set automatically by setuptools.

Maybe this issue is a duplicate of https://github.com/pypa/setuptools/issues/1283, I'm not sure.

**Comment by jaraco** [comment]
@dan-santana I can't tell immediately if this is a duplicate or not, but the same guidance may apply. Can you use `pip install .` to install the package and if you do does it have the same issue?

**Comment by dan-santana** [comment]
Yes, using `pip install .` still gives me the wrong permissions for the `<package>.egg-info/*` files.

`ls -lh /usr/lib/python3.6/site-packages/sshuttle-0.78.4-py3.6.egg-info/ `:
```
total 24K
-rw------- 1 root root    1 Apr 19 01:20 dependency_links.txt
-rw------- 1 root root   52 Apr 19 01:20 entry_points.txt
-rw-r--r-- 1 root root 1.8K Apr 19 01:20 installed-files.txt
-rw------- 1 root root 3.1K Apr 19 01:20 PKG-INFO
-rw------- 1 root root 1.5K Apr 19 01:20 SOURCES.txt
-rw------- 1 root root    9 Apr 19 01:20 top_level.txt
```

**Comment by jaraco** [comment]
I'd be happy to see this fixed, but it still needs more analysis to identify where the failed expectation lies. Is it in the generation of the egg-info? Or in how it's copied to the build/installation target? How does one recreate and identify the issue?

**Comment by lilydjwg** [comment]
[Telethon had similar issues](https://github.com/LonamiWebs/Telethon/issues/1605) too. It made many (if not all) Python programs fail due to unreadable files. It also has several generated world-writable files.

What's more, when one extracts the sdist tarball as root, the files are owned by whatever  uid and gid the tarball has recorded.

Is it possible to give all files in the created tarballs 755 / 644 permissions and owned by root:root?
